src/wwwroot/app/directives/candidate-vacancy-info/candidate-vacancy-info.js

```diff
@@ -29,179 +35,229 @@ export default class CandidateVacancyInfoDirective {
    }
 }
 
-function CandidateVacancyInfoController($scope, $translate, $q, $state, UserDialogService) {
+function CandidateVacancyInfoController($scope, // eslint-disable-line max-statements
+      UserDialogService,
+      $translate,
+      $state,
+      $q
+) {
    'ngInject';
    const vm = $scope;
+   const STAGE_STATES = {
+      'Inactive': 0,
+      'Active': 1,
+      'Passed': 2,
+      'NotPassed': 3
+   };
+   const STAGE_TYPES = {
+      'MainStage' : 1,
+      'HireStage' : 2,
+      'RejectStage' : 3
+   };
+   vm.stagesToShow = [];
+   vm.vacancyStageInfoComposedObjectsToShow = vm.vacancyStageInfosComposedByCandidateIdVacancyId;
+   vm.stageQueries = [];
 
-   function handleStagePassing(stagesObject, stageToPass) {
-      let deffered = $q.defer();
-      if (stageToPass.isCommentRequired) {
-         showSingleCommentDialog().then((dialogTransferObject) => {
-            dialogTransferObject.isPassed = true;
-            deffered.resolve({ comment: dialogTransferObject.comment,
-               isPassed: dialogTransferObject.isPassed,
-               stagesObject,
-               stageToPass });
-         }).catch(() => {
-            deffered.reject();
+   function _init() {
+      if (vm.parentEntity === 'vacancy') {
+         vm.rejectStages = filter(vm.vacancyStages, (extStage) => {
+            return extStage.stage.stageType === STAGE_TYPES.RejectStage;
          });
-      } else {
-         deffered.resolve({ comment: null,
-            isPassed: true,
-               stagesObject,
-               stageToPass  });
+         vm.stagesToShow = calculateVacancyStagesEntitiesCount(
+               vm.vacancyStageInfosComposedByCandidateIdVacancyId);
+      } else if (vm.parentEntity === 'candidate') {
+         vm.vacancyStageInfosComposedByCandidateIdVacancyId = map(
+               vm.vacancyStageInfosComposedByCandidateIdVacancyId, findAndSetRejectStagesFor);
+         if (some(vm.vacancyStageInfosComposedByCandidateIdVacancyId)) {
+            //TODO intersect of stages
+            vm.stagesToShow = calculateVacancyStagesEntitiesCount(vm.vacancyStageInfosComposedByCandidateIdVacancyId);
+         }
       }
-      return deffered.promise;
    }
 
-   function showSingleCommentDialog() {
+   function findAndSetRejectStagesFor(vacancyStage)  {
+      return Object.assign (vacancyStage, {
+         rejectStages: filter(vacancyStage.stageFlow, (extStage) => {
+            return extStage.stage.stageType === STAGE_TYPES.RejectStage;
+         })
```

Comment:
it can be next:
rejectStages: filter(vacancyStage.stageFlow, extStage => extStage.stage.stageType === STAGE_TYPES.RejectStage);